dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/FileResourceStream.java

```diff
@@ -43,5 +53,96 @@
 @RequiredArgsConstructor
 public class FileResourceStream {
   private final FileResource fileResource;
-  @Setter private FileResourceSupplier<InputStream> inputStreamSupplier;
+  private FileResourceSupplier<InputStream> inputStreamSupplier;
+
+  @Nonnull
+  public static FileResourceStream of(
+      FileResourceService fileResourceService, FileResource fileResource) {
+    return new FileResourceStream(
+        fileResource,
+        () -> {
+          try {
+            return fileResourceService.openContentStream(fileResource);
+          } catch (NoSuchElementException e) {
+            // Note: we are assuming that the file resource is not available yet. The same approach
+            // is taken in other file endpoints or code relying on the storageStatus = PENDING.
+            // All we know for sure is the file resource is in the DB but not in the store.
+            throw new ConflictException(
+                "The content is being processed and is not available yet. Try again later.");
+          } catch (IOException e) {
```

Comment:
Would it be worth moving those strings as constants? I see we have in quite a few places.

Reply:
I don't want to adapt code other than tracker right now. For tracker they are contained in this class.

We will work on improving the serving of files in general across products after the next release 😄